src/core.rs

```diff
@@ -87,6 +90,162 @@ pub struct WormholeWelcome {
  * Maybe a better way to handle application level protocols is to create a trait for them and then
  * to paramterize over them.
  */
+
+pub struct MailboxConnection<V: serde::Serialize + Send + Sync + 'static> {
+    config: AppConfig<V>,
+    server: RendezvousServer,
+    pub welcome: Option<String>,
+    pub mailbox: Mailbox,
+    pub code: Code,
+}
+
+impl<V: serde::Serialize + Send + Sync + 'static> MailboxConnection<V> {
+    ///
+    /// Create a connection to a mailbox which is configured with a `Code` starting with the nameplate and by a given number of wordlist based random words.
+    ///
+    /// # Arguments
+    ///
+    /// * `config`: Application configuration
+    /// * `code_length`: number of words used for the password. The words are taken from the default wordlist.
+    ///
+    /// returns: Result<MailboxConnection<V>, WormholeError>
+    ///
+    /// A `MailboxConnection` contains the a `RendezvousServer` which is connected to the mailbox which has been created and the following parameters:
+    ///
+    /// A copy of `AppConfig`,
+    /// the `RendezvousServer` with the open mailbox connection,
+    /// the welcome message received from the mailbox server,
+    /// the mailbox id and
+    /// the Code which is required to connect to the mailbox.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::executor;
```

Comment:
It's better to try and hide the setup stuff which is not directly linked to the usage. The example should be easy to copy-paste, and for that we may assume that the developer is already in an appropriate context (in this case, async and returning a `Result`).

You can have hidden setup lines by prefixing them with `#`, and the `async_std` crate does the following for their examples (I just adapted the Result type):

```suggestion
    /// # fn main() -> eyre::Result<()> { async_std::task::block_on(async {
```

(Full reference: https://docs.rs/async-std/latest/src/async_std/net/tcp/stream.rs.html#62-72)

Reply:
Fully agree here. Thanks for the hint. This works great.